api/analyzedphenotypes.cv.api.inc

```diff
@@ -95,196 +102,180 @@ function analyzedphenotypes_cvprop($property = null, $parameter = null) {
-  // search is not limited to an controlled vocabulary (cv).
+  // FROM.
+  $sql .= 'FROM';
+  $sql_tbl = '{cvterm}';
   if ($value == 'cvtermcv') {
-    $cvterm = chado_query(
-      "SELECT cvterm_id FROM {cvterm} AS t1 INNER JOIN {cv} AS t2 USING(cv_id)
-      WHERE CONCAT(t1.name, ' (' , t2.name , ')') = :value LIMIT 1",
-      array(':value' => $identifiers['cvtermcv'])
-    );
-
-    return ($cvterm->rowCount()) ? $cvterm->fetchField() : null;
+    $sql_tbl .= ' INNER JOIN {cv} USING(cv_id)';
   }
 
-  // # SEARCH BY CV ID.
-  // search is limited to controlled vocabulary (cv) provided.
-  if ($value == 'cv_id') {
-    $cvterm = chado_query(
-      "SELECT cvterm_id, name, definition FROM {cvterm}
-      WHERE cv_id = :value ORDER BY name ASC",
-      array(':value' => $identifiers['cv_id'])
-    );
+  $sql .= ' ' . $sql_tbl . ' ';
 
-    if ($cvterm) {
-      $term = array();
+  // WHERE.
+  $sql .= 'WHERE';
+  $sql_where = array('field', 'match' => '=', 'value', 'args' => ':value');
 
-      foreach($cvterm as $c) {
-        $term[$c->cvterm_id] = array(
-          'name' => $c->name,
-          'definition' => $c->definition,
-        );
-      }
-
-      return $term;
-    }
+  if ($value == 'cvterm_id') {
+    // BY CVTERM ID NUMBER.
+    $sql_where['field'] = 'cvterm.cvterm_id';
+    $sql_where['value'] = $identifiers[$value];
+  }
+  elseif ($value == 'name') {
+    // BY CVTERM NAME.
+    $sql_where['field'] = 'cvterm.name';
+    $sql_where['value'] = $identifiers[$value];
+  }
+  elseif ($value == 'keyword') {
+    // BY CVTERM NAME KEYWORD.
+    $sql_where['field'] = 'LOWER(cvterm.name)';
+    $sql_where['match'] = 'LIKE';
+    $sql_where['value'] = '%' . strtolower($identifiers[$value]) . '%';
+  }
+  elseif ($value == 'cvtermcv') {
+    // BY CVTERM NAME AND CV NAME COMBINATION.
+    $sql_where['field'] = 'CONCAT(cvterm.name, \' (\', cv.name, \')\')';
+    $sql_where['value'] = $identifiers[$value];
+  }
+  elseif ($value == 'cv_id') {
+    // BY CV ID NUMBER.
+    $sql_where['field'] = 'cvterm.cv_id';
+    $sql_where['value'] = $identifiers[$value];
   }
 
-  // May request detailed cvterm properties when implied in $options parameter.
+  // field = :value
+  $sql .= ' ' . $sql_where['field'] . ' ' . $sql_where['match'] . ' ' . $sql_where['args'] . ' ';
 
-  // # SEARCH BY NAME.
-  // search is limited to the genus cv configuration.
-  if ($value == 'name') {
-    $args = array(
-      'cv_id' => $sysvar_genus_cv,
-      'name' => $identifiers['name']
+  // WHERE AND.
+  if (isset($identifiers['genus'])) {
+    // Restrict dataset to the cv configuration for a genus.
+    $sysvar = ap_get_variablenames(
+      array('variablename' => $identifiers['genus']),
+      array('set' => 'cvdbon', 'suffix' => 'cv')
     );
 
-    $cvterm = (function_exists('chado_get_cvterm')) ? chado_get_cvterm($args) : tripal_get_cvterm($args);
+    $sysvar_genus_cv = variable_get($sysvar);
+    $sql .= 'AND cvterm.cv_id = ' . $sysvar_genus_cv . ' ';
```

Comment:
You should never embed a value directly in the query... Instead perhaps you should have a "where args" array like I did in my example.